db.go

```diff
@@ -213,6 +215,76 @@ func (db *DB) Get(ro *ReadOptions, key []byte) ([]byte, error) {
 	return C.GoBytes(unsafe.Pointer(value), C.int(vallen)), nil
 }
 
+// GetMany returns the values associated with keys from the database.
+//
+// GetMany fills in the entries of values consistent with Get():
+// - If keys[i] exists in the database with non-zero value, values[i] holds its non-zero len []byte slice.
+// - If keys[i] does not exist in the database, values[i] == nil.
+// - If keys[i] does exist, but its value is zero-length, values[i] == []byte{}.
+// - If there's an error looking up keys[i], values[i] == nil.
+//
+// The keys byte slice may be reused safely. Go takes a copy of
+// them before returning.
+func (db *DB) GetMany(ro *ReadOptions, keys [][]byte) ([][]byte, []error) {
+	if db.closed {
+		panic(ErrDBClosed)
+	}
+	if len(keys) == 0 {
+		return nil, nil
+	}
+
+	var packedKeysBuffer bytes.Buffer
+	cKeyLens := make([]C.size_t, len(keys))
+	for i := range keys {
+		packedKeysBuffer.Write(keys[i])
+		cKeyLens[i] = C.size_t(len(keys[i]))
+	}
+	packedKeysBytes := packedKeysBuffer.Bytes()
+
+	var cPackedVals *C.char
+	var cValLens *C.size_t
+	var cPackedErrs *C.char
+	var cErrLens *C.size_t
+	C.leveldb_getmany(
+		db.Ldb, ro.Opt, (*C.char)(unsafe.Pointer(&packedKeysBytes[0])), C.size_t(len(keys)), &cKeyLens[0],
+		&cPackedVals, &cValLens,
+		&cPackedErrs, &cErrLens)
+	defer C.leveldb_free(unsafe.Pointer(cPackedErrs))
+	defer C.leveldb_free(unsafe.Pointer(cErrLens))
+	defer C.leveldb_free(unsafe.Pointer(cPackedVals))
+	defer C.leveldb_free(unsafe.Pointer(cValLens))
+
+	// Unpack the errors from C chars into Golang error objects
+	errLens := (*[1 << 30]C.size_t)(unsafe.Pointer(cErrLens))[:len(keys):len(keys)]
+	errs := make([]error, len(keys))
+	offset := 0
+	for i := range errLens {
+		if errLens[i] == 0 {
+			continue
+		}
+		b := C.GoBytes(unsafe.Pointer(uintptr(unsafe.Pointer(cPackedErrs))+uintptr(offset)), C.int(errLens[i]))
+		errStrLen := int(errLens[i])
+		errs[i] = fmt.Errorf(string(b[:errStrLen]))
+		offset += errStrLen
+	}
+
+	// Unpack the packed values from C chars into Golang byte slices
+	valueLens := (*[1 << 30]C.size_t)(unsafe.Pointer(cValLens))[:len(keys):len(keys)]
+	values := make([][]byte, len(keys))
+	offset = 0
+	for i := range valueLens {
+		if valueLens[i] > 0 {
+			values[i] = C.GoBytes(unsafe.Pointer(uintptr(unsafe.Pointer(cPackedVals))+uintptr(offset)), C.int(valueLens[i]))
+			offset += int(valueLens[i])
+		} else if errs[i] == nil && valueLens[i] == 0 {
+			// No error and no value, it means keys[i] is found but with empty value
+			values[i] = []byte{}
+		}
```

Comment:
nit: can we explicitly document that `errs[i] == nil && valueLens[i] == -1` means key is not found ? 

(it's clear from: https://github.com/DataDog/leveldb/pull/10/files#diff-c53e6c8e4b8bd615409f4d877ccb4e96R251 but you have to read the c code to understand)